src/scripts/main.js

```diff
@@ -1,3 +1,243 @@
+  }
+
+  const tableBody = [...table.rows].slice(1, -1).sort((a, b) => {
+    let firstValue = a.cells[currentPos].innerHTML;
+    let secondValue = b.cells[currentPos].innerHTML;
+
+    // WE CHECK WHETHER THESE ARE NUMBERS
+    if (!isNaN(parseFloat(firstValue)) || firstValue.includes('$')) {
+      if (firstValue.includes('$')) {
+        firstValue = firstValue.slice(1).replace(',', '.');
+        secondValue = secondValue.slice(1).replace(',', '.');
+      }
+
+      return +firstValue - +secondValue;
+    }
+
+    return firstValue > secondValue ? 1 : -1;
+  });
+
+  if (elem.dataset.sort === 'desc') {
+    table.tBodies[0].append(...tableBody.reverse());
+  } else {
+    table.tBodies[0].append(...tableBody);
+  }
+}
+
+function createNewForm() {
+  const formEmployee = document.createElement('form');
+  const button = document.createElement('button');
+
+  button.setAttribute('type', 'submit');
+  button.textContent = 'Save to table';
+
+  formEmployee.classList.add('new-employee-form');
+  formEmployee.append(createInput('name', 'text', 'input'));
+  formEmployee.append(createInput('position', 'text', 'input'));
+
+  formEmployee.append(
+    createInput('office', '', 'select', [
+      'Tokyo',
+      'Singapore',
+      'London',
+      'New York',
+      'Edinburgh',
+      'San Francisco',
+    ]),
+  );
+  formEmployee.append(createInput('age', 'number', 'input'));
+  formEmployee.append(createInput('salary', 'number', 'input'));
+  formEmployee.append(button);
+  wrapperForm.append(formEmployee);
+  wrapperForm.style.alignSelf = 'flex-start';
+  document.body.append(wrapperForm);
+}
+
+createNewForm();
+
+function createInput(value, type, typeInput, array = null) {
+  const label = document.createElement('label');
+  const input = document.createElement(typeInput);
+
+  input.setAttribute('name', value);
+  input.setAttribute('required', '');
+  input.setAttribute('data-qa', value);
+
+  if (typeInput === 'select') {
+    array.forEach((elem) => {
+      const option = document.createElement('option');
+
+      option.setAttribute('value', elem);
+      option.textContent = elem;
+      input.append(option);
+    });
+  }
+
+  label.textContent = `${value[0].toUpperCase()}${value.slice(1)}:`;
+
+  label.append(input);
+
+  input.setAttribute('type', type);
+
+  return label;
+}
+
+const form = document.querySelector('form');
+const fullName = form.querySelector('[data-qa="name"]');
+const position = form.querySelector('[data-qa="position"]');
+const office = form.querySelector('[data-qa="office"]');
+const age = form.querySelector('[data-qa="age"]');
+const salary = form.querySelector('[data-qa="salary"]');
+
+// ADD NEW EMPLOYEE IN THE TABLE
+form.addEventListener('submit', addNewEmployeeInTable);
+
+function addNewEmployeeInTable(e) {
+  e.preventDefault();
+
+  if (!checkValidate()) {
+    return;
+  }
+
+  const newRow = document.createElement('tr');
+
+  const salaryChange = `$${salary.value},000`;
```

Comment:
The salary value is being formatted as a string with a comma, which might cause issues if further calculations are needed. Consider storing it as a number and formatting it only when displaying.